library/src/main/java/com/novoda/downloadmanager/LiteBatchFileBuilder.java

```diff
@@ -40,20 +40,60 @@ public BatchFileBuilder saveTo(String path) {
 
     @Override
     public BatchFileBuilder saveTo(String path, String fileName) {
-        this.path = Optional.fromNullable(path).filterNot("/"::equals);
+        this.path = Optional.fromNullable(path);
         this.fileName = Optional.fromNullable(fileName);
         return this;
     }
 
     @Override
     public BatchBuilder apply() {
-        String absolutePath = storageRoot.path() + File.separator
-                + downloadBatchId.rawId() + File.separator
-                + path.map((it) -> it + File.separator).or("")
-                + fileName.getOrElse(() -> FileNameExtractor.extractFrom(networkAddress));
+        String absolutePath = buildPath(
+                storageRoot.path(),
+                downloadBatchId.rawId(),
+                path.or(""),
+                fileName.getOrElse(() -> FileNameExtractor.extractFrom(networkAddress))
+        );
 
         parentBuilder.withFile(new BatchFile(networkAddress, downloadFileId, absolutePath));
         return parentBuilder;
     }
 
+    private String buildPath(String... elements) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        if (elements[0].startsWith(File.separator)) {
+            stringBuilder.append(File.separator); // if storage path starts with /, we honour it
+        }
+
+        for (String element : elements) {
+            if (element.isEmpty()) {
+                continue; // ignore empty paths
```

Comment:
`continue` == `goto` :trollface:

Reply:
😄 I would have used `filter` but you know, Java.